### PR TITLE
Document that we conform to SPEC0 version support.

### PR DESCRIPTION
**Pull request by samcunliffe**
(Finally) closes 
- #362

as per the definition of done in @dstansby's last comment.

Think this is fairly uncontroversial (the decision itself is already logged in #362).

### CODE REVIEW

**Comment by samcunliffe** [comment]
Good guy @pre-commit.ci

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -53,6 +53,14 @@ cookiecutter /path/to/your/checkout/of/python-tooling --checkout <branch-name>
 You can omit the `--checkout` option if you're already on the
 branch you want to test.
 
+#### Python version support
+
+[![SPEC 0 — Minimum Supported Dependencies](https://img.shields.io/badge/SPEC-0-green?labelColor=%23004811&color=%235CA038)](https://scientific-python.org/specs/spec-0000/)
+
+We follow Scientific Python's [SPEC0](https://scientific-python.org/specs/spec-0000/) policy.
```
Comment (dstansby) [inline CONTRIBUTING.md]:
```suggestion
We follow Scientific Python's [SPEC 0](https://scientific-python.org/specs/spec-0000/) policy.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -53,6 +53,14 @@ cookiecutter /path/to/your/checkout/of/python-tooling --checkout <branch-name>
...
+Notably this means our template will support [Python versions for three years after their release](https://scientific-python.org/specs/spec-0000/#support-window), and changes to the template are equired to pass tests in all such python versions.
```
Comment (samcunliffe) [inline CONTRIBUTING.md]:
```suggestion
Notably this means our template will support [Python versions for three years after their release](https://scientific-python.org/specs/spec-0000/#support-window), and changes to the template are required to pass tests in all such python versions.
```

Comment (dstansby) [inline CONTRIBUTING.md]:
```suggestion
Notably this means our template will support [Python versions for three years after their release](https://scientific-python.org/specs/spec-0000/#support-window), and changes to the template are required to pass tests on all such python versions.
```

**Review by dstansby — APPROVED** [review]

**Pull request merged**